### ISSUE TITLE
No longer able to pass bucket name as empty string

### ISSUE THREAD

**Issue by matthewduren** [terraform-providers/terraform-provider-aws#7868]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.11
AWS provider v2.1.0

### Affected Resource(s)

* aws_s3_bucket

### Terraform Configuration Files

```hcl
variable "bucket_name" {
  default = ""
}

resource "aws_s3_bucket" "main" {
  bucket = "${var.bucket_name}"
....
}
```

### Expected Behavior

The bucket should have been given an auto-generated name

### Actual Behavior

Length validation error

```
aws_s3_bucket.main: expected length of bucket to be in the range (3 - 63), got 

```

### References

See https://github.com/terraform-providers/terraform-provider-aws/pull/7778

Offending line: 
https://github.com/terraform-providers/terraform-provider-aws/pull/7778/files#diff-ec071549f7daf514c0a6203559384e80R42
```
ValidateFunc:  validation.StringLenBetween(3, 63),
```

**Comment by bflad** [comment]
Fix submitted: #7881

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.2.0 of the Terraform AWS Provider, likely later today.

**Comment by bflad** [comment]
This has been released in [version 2.2.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#220-march-15-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
